net.py

```diff
@@ -1,172 +0,0 @@
-        dW = 1/m * np.dot(dZ, self.A_saved[self.L-1].T)
-        db = 1/m * np.sum(dZ, axis=1, keepdims=True)
-        dA = np.dot(self.W[self.L].T, dZ)
-        self.dZ[self.L] = dZ
-        self.dW[self.L] = dW
-        self.db[self.L] = db
-        self.dA[self.L] = dA
-        for i in reversed(range(1,self.L)):
-            dZ = np.array(dA, copy=True)
-            #print('dZ.shape ' + str(dZ.shape))
-            #print('Z.shape ' + str(self.Z_saved[i].shape))
-            dZ[self.Z_saved[i] <= 0] = 0
-            #print('i: ' + str(i))
-            dW = 1/m * np.dot(dZ, self.A_saved[i-1].T)
-            #print('dW.shape ' + str(dW.shape))
-            #print('W.shape ' + str(self.W[i].shape))
-            #print('A.shape ' + str(self.A_saved[i].shape))
-            db = 1/m * np.sum(dZ, axis=1, keepdims=True)
-            dA = np.dot(self.W[i].T,dZ)
-            self.dZ[i] = dZ
-            self.dW[i] = dW
-            self.db[i] = db
-            self.dA[i] = dA
-
-    def parameters_update(self, learning_rate):
-        for i in range(1,self.L+1):
-            #print(self.W[i].shape)
-            #print(self.dW[i].shape)
-            self.W[i] = self.W[i] - learning_rate * self.dW[i]
-            self.b[i] = self.b[i] - learning_rate * self.db[i]
-
-    def train(self, X, Y, learning_rate, iter_no):
-        self.A_saved[0] = X
-        for i in range(iter_no):
-            self.forward_prop(X)
-            cost = self.cost_function(Y)
-            self.backward_prop(Y)
-            self.parameters_update(learning_rate)
-            print(cost)
-
-    def grad_check_change_values(self, change=False):
-        if not change:
-            last_index = 0
-            # skip first element (-1)
-            W_iter = iter(self.W)
-            next(W_iter)
-            for i, w in enumerate(self.W):
-                size_x, size_y = self.layers[i-1], self.layers[i]
-                next_index = size_x * size_y + last_index
-                w = np.reshape(self.backup[last_index:(last_index+size_x*size_y)], (size_x,size_y))
-                last_index = next_index
-                #next_index =
-        else:
-            flatten = lambda l: [item for sublist in l for item in sublist]
-            backup = flatten(self.W) + flatten(self.b)
-            self.W = self.W + change
-
-    def grad_check(self, X, Y, epsilon = 1e-7):
-        self.A_saved[0] = X
-        self.forward_prop(X)
-        #print(self.W)
-        self.grad_check_change_values(epsilon)
-
-
-if __name__ == '__main__':
-    #np.seterr(divide='ignore', invalid='ignore')
-    #np.random.seed(1)
-    test1 = Net((8, 5, 1))
-    #X = np.array([0.05, 11, 100, 2, 0.8], ndmin=2)
-    #X = np.random.randn(10, 5)
-    #Y = np.random.randint(2, size=(1, 10))
-    #print(X)
-    data = np.genfromtxt('pima-indians-diabetes.csv', delimiter=',')
-    Y = np.array(data[:,-1], ndmin=2)
-    X = data[:,0:-1]
-    X = X.T
-    #test1.grad_check(X,Y)
-    print(test1.layers)
```

Comment:
If you want to develop this module in future, think about using of logging instead prints. https://docs.python.org/3/library/logging.html 
Define debug.py module with logging config and import it. Then just set right level (in development debug, in release info)